lib/src/main/java/com/iwillfailyou/inspections/inheritancefree/inheritances/JavaInheritances.java

```diff
@@ -0,0 +1,127 @@
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public final class JavaInheritances implements Violations<Inheritance> {
+
+    private final Violations<Inheritance> inheritances;
+
+    public JavaInheritances(final String... lines) {
+        this(ParserConfiguration.LanguageLevel.RAW, lines);
+    }
+
+    public JavaInheritances(final ParserConfiguration.LanguageLevel level, final String... lines) {
+        this(
+            new JavaParser(new ParserConfiguration().setLanguageLevel(level)),
+            lines
+        );
+    }
+
+    public JavaInheritances(final JavaParser parser, final String... lines) {
+        this(
+            parser,
+            new InspectionScalar<>(() -> {
+                final StringBuilder source = new StringBuilder();
+                for (final String line : lines) {
+                    source.append(line);
+                }
+                return new ByteArrayInputStream(source.toString().getBytes());
+            }),
+            String.join("", lines)
+        );
+    }
+
+    public JavaInheritances(final String source) {
+        this(ParserConfiguration.LanguageLevel.RAW, source);
+    }
+
+    public JavaInheritances(final ParserConfiguration.LanguageLevel level, final String source) {
+        this(
+            new JavaParser(new ParserConfiguration().setLanguageLevel(level)),
+            source
+        );
+    }
+
+    public JavaInheritances(final JavaParser parser, final String source) {
+        this(
+            parser,
+            new InspectionScalar<>(() -> new ByteArrayInputStream(source.getBytes())),
+            source
+        );
+    }
+
+    public JavaInheritances(final File file) {
+        this(ParserConfiguration.LanguageLevel.RAW, file);
+    }
+
+    public JavaInheritances(final ParserConfiguration.LanguageLevel level, final File file) {
+        this(
+            new JavaParser(new ParserConfiguration().setLanguageLevel(level)),
+            file
+        );
+    }
+
+    public JavaInheritances(final JavaParser parser, final File file) {
+        this(
+            parser,
+            new InspectionScalar<>(() -> {
+                try {
+                    return new FileInputStream(file);
+                } catch (final IOException e) {
+                    throw new InspectionException(
+                        String.format(
+                            "Can not get an input stream from the file: %s",
+                            file
+                        ),
+                        e
+                    );
+                }
+            }),
+            file.getAbsolutePath()
+        );
+    }
+
+    public JavaInheritances(
+        final JavaParser parser,
+        final InspectionScalar<InputStream> source,
+        final String descriptor
+    ) {
+        this(
+            new JavaViolations<>(
```

Comment:
Let's implement it other way, see the [`NoMultipleReturn`](https://github.com/iwillfailyou/java-inspections/blob/master/lib/src/main/java/com/iwillfailyou/inspections/nomultiplereturn/NoMultipleReturn.java) and [`MultipleReturnViolations`](https://github.com/iwillfailyou/java-inspections/blob/master/lib/src/main/java/com/iwillfailyou/inspections/nomultiplereturn/multiplereturns/MultipleReturnViolations.java) - it has more flexible approach